File: python-playwright/src/pages/page_dropdown.py
# Created on : Apr 2, 2026, 10:19:27 AM
# Author     : Tahreem J. Naseem (www.tahreems.com / +92-333-443-8775) {TECHBYTES}

from src.core.base_page import BasePage
from src.utilities.logger import get_logger


class PageDropdown(BasePage):
    logger = get_logger(__name__)

    PAGE_PATH = "/dropdown"
    DROPDOWN = "#dropdown"

    def open(self, base_url: str):
        self.logger.info(f"Opening Dropdown page at {base_url}{self.PAGE_PATH}")
        self.navigate(f"{base_url}{self.PAGE_PATH}")

    def select_value(self, value: str):
        self.logger.info(f"Selecting value '{value}' from dropdown.")
        self.select_dropdown(self.DROPDOWN, value)

    def get_selected_value(self) -> str:
        self.logger.info("Getting selected value from dropdown.")
        text = self.get_text_selected_dropdown(self.DROPDOWN)
        if text is None:
            self.logger.error("'None' data type recieved from dropdown.")
            raise ValueError("'None' data type recieved from dropdown.")
        return text.strip()

File: javascript-playwright/src/pages/page-dropdown.js
import { BasePage } from "../core/base-page.js";

export class PageDropdown extends BasePage {
  PAGE_PATH = "/dropdown";
  DROPDOWN = "#dropdown";

  async open(baseUrl) { await this.navigate(`${baseUrl}${this.PAGE_PATH}`); }
  async selectValue(value) { await this.selectDropdown(this.DROPDOWN, value); }
  async getSelectedValue() { return (await this.getTextSelectedDropdown(this.DROPDOWN)).trim(); }
}
